R/AddCardModule.R

```diff
@@ -1,10 +1,47 @@
-#' Add Card Button User Interface
+#' Add card button module
+#'
 #' @description `r lifecycle::badge("experimental")`
-#' button for adding views/cards to the Report.
+#'
+#' Provides UI and server functions to add views/cards to a report.
 #'
 #' For more details see the vignette: `vignette("simpleReporter", "teal.reporter")`.
+#'
+#' @details
+#' The `card_fun` function is designed to create a new `ReportCard` instance and optionally customize it:
+#' - The `card` parameter allows for specifying a custom or default `ReportCard` instance.
+#' - Use the `comment` parameter to add a comment to the card via `card$append_text()`- if `card_fun` does not
+#' have `comment` parameter, then `comment` from `Add Card UI` module will be added at the end of the content of the
+#' card.
+#' - The `label` parameter enables customization of the card's name and its content through `card$append_text()`-
+#' if `card_fun` does not have `label` parameter, then `card name` will be set to the name passed in
+#' `Add Card UI` module, but no text will be added to the content of the `card`.
+#'
+#' This module supports using a subclass of [`ReportCard`] for added flexibility.
+#' A subclass instance should be passed as the default value of
+#' the `card` argument in the `card_fun` function.
+#' See below:
+#' ```{r}
+#' CustomReportCard <- R6::R6Class( # nolint: object_name_linter.
+#'   classname = "CustomReportCard",
+#'   inherit = teal.reporter::ReportCard
+#' )
+#'
+#' custom_function <- function(card = CustomReportCard$new()) {
+#'   card
+#' }
+#' ```
+#' @name add_card_button
+#'
 #' @param id `character(1)` this `shiny` module's id.
-#' @return `shiny::tagList`
+#' @param reporter [`Reporter`] instance.
+#' @param card_fun `function` which returns a [`ReportCard`] instance.
+#' It can have optional `card`, `comment` and
+#' `label` parameters.
+#'
+#' @return `shiny::moduleServer`.
```

Comment:
No, it doesn't the `moduleServer` function.

Reply:
since the doc is grouped together between ui and server for add card module, I guess technically it's correct that it doesn't return **just** `moduleServer`. 

The add card module ui returns `tagList` and the server returns `modulesServer`.

Reply:
No, it doesn't. It returns the value of a `moduleServer` call, which, as far as I can tell, is `NULL`.

Reply:
I agree. How about this?
```
Return `shiny.tag` in the case of `add_card_button_ui` and the return value, if any, from executing the module server function in case of `add_card_button_srv`.
```

Reply:
In a Shiny module's UI function, it is customary to return an object of class` shiny.tag` or shiny.tag.list. Mentioning the UI component specifically in the documentation I believe is not necessary and I think I removed  this (UI return element)  in other packages based aleks recommendation before.

 as the focus often lies on the server logic encapsulated by moduleServer.  given that the UI's role is implicitly understood to provide the necessary user interface elements.

Reply:
done [f8cc1b3](https://github.com/insightsengineering/teal.reporter/pull/247/commits/f8cc1b3d433d64dc79893509ab22e27295047eeb)

Reply:
We can skip the fact that the UI function returns a `shiny.tag` because every UI function does that and the value is never used. It is the server value that matters because it may have a purpose.